Few changes in installation

frankmayer opened the pull request:
### Summary of Changes
- Unnecessary double quotes
- `dirname()` to `__DIR__`
- Optimized loop termination condition
- Short syntax for applied operation
- Remove one-time-use variable
- Remove superfluous property initialization

The changes in this PR should be fairly easy to review. In hope that this will get merged quickly so further work can be done without conflicting with other PRs. ;)

### Testing Instructions

None, should not change behavior


### Documentation Changes Required

None.

frankmayer committed:
Unnecessary double quotes

frankmayer committed:
dirname to __DIR__

frankmayer committed:
Optimized loop termination condition

frankmayer committed:
Short syntax for applied operation

frankmayer committed:
Remove one-time-use variable

frankmayer committed:
Remove superfluous property initialization

mbabker commented on installation/index.php:
```diff
@@ -23,7 +23,7 @@
 
 // Bootstrap the application
-require_once dirname(__FILE__) . '/application/bootstrap.php';
+require_once __DIR__ . '/application/bootstrap.php';
```
Please revert this one, same reasoning as the comment a few lines above.

frankmayer replied:
@mbabker which lines are you referring to? 

Doesn't this apply? http://php.net/manual/en/language.constants.predefined.php

frankmayer replied:
Yes, got it. This old stuff continues biting us... :smile: 
But, who uses 5.2?!?!?!?!
I'd say with Joomla 4 this should change without mercy :smile_cat:

frankmayer replied:
Reverted...

frankmayer replied:
Will take a look at that PR. Sounds great!

frankmayer committed:
DO NOT REPLACE dirname() with __DIR__ in installation because of B/C with 5.2!!!
DO NOT REPLACE dirname() with __DIR__ in installation because of B/C with 5.2!!!
DO NOT REPLACE dirname() with __DIR__ in installation because of B/C with 5.2!!!
DO NOT REPLACE dirname() with __DIR__ in installation because of B/C with 5.2!!!

RonakParmar commented on installation/model/database.php:
```diff
@@ -1213,7 +1213,7 @@ protected function splitQueries($query)
 
 		// Add function part as is.
-		for ($f = 1; $f < count($funct); $f++)
+		for ($f = 1, $fMax = count($funct); $f < $fMax; $f++)
```
No need to change this for loop below code works fine.
for ($f = 1; $f < count($funct); $f++)

frankmayer replied:
Actually there is, because as it it now, in every iteration, the `$funct` is unnecessarily counted again and again.
With my change, it is counted once and the result used in every iteration, for the comparison.

RonakParmar replied:
I googled about this and found bellow code to optimize for loop.
$fMax = (int) count($funct);
for ($f = 1; $f < $fMax; $f++)

frankmayer replied:
It's essentially the same thing. In my change it is a one-liner. (The (int) is unnecessary, because count returns an integer anyway. See: http://php.net/manual/en/function.count.php)

RonakParmar replied:
Okay, Thanks for sharing your knowledge. :+1:

RonakParmar commented:
I have tested this item :white_check_mark: successfully on a12727e8a965a4114e7d43f4db88c3b22166318e<br /><br />Done code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13297">issues.joomla.org/tracker/joomla-cms/13297</a>.</sub>

shur commented:
I have tested this item :white_check_mark: successfully on a12727e8a965a4114e7d43f4db88c3b22166318e<br /><br />Code Review.
I approve it here because approval on github doesn't count for some unknown reason.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13297">issues.joomla.org/tracker/joomla-cms/13297</a>.</sub>

jeckodevelopment commented:
@shur code review doesn't allow see errors while "using" the code.

jeckodevelopment commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13297">issues.joomla.org/tracker/joomla-cms/13297</a>.</sub>
